### ISSUE TITLE
In proposals function should be before function reference

### ISSUE THREAD

**Issue by paulvi** [angelozerr/tern.java#333]
In proposals function should be before function reference

![image](https://cloud.githubusercontent.com/assets/1614482/10097978/f6ccd198-63af-11e5-8d9a-7bcdc17fad38.png)

because usually we are selecting function.

**Comment by angelozerr** [comment]
@paulvi I have done this choice because when you have several signature, the function is reference is on the bottom and it's not easy to select it.

Perhaps it could be a preferences?

**Comment by jabby** [comment]
I agree with @paulvi. A preference to set this is a good idea. 

Maybe it should be set by default with function before reference ?

**Comment by paulvi** [comment]
@angelozerr 

Just to clarify, it should be as

```
start()
start
stop()
stop
```

unfortunately that is not natural order.

And possibly user needs get used to press down....

**Comment by paulvi** [comment]
BTW, `:` goes after `(` in ASCII http://www.asciitable.com/

so removing space in `start : fn` would solve

**Comment by angelozerr** [comment]
Yes I have understood, but if you take jQuery animation, the referenced function is on the bottom:

![jqueryanimate](https://cloud.githubusercontent.com/assets/1932211/10106989/29987b2e-63b8-11e5-880a-28b6b54d56d6.png)

@vrubezhny @dgolovin @mickaelistria @maxandersen @pascalleclercq @gamerson @fbricon @piotrtomiak what do you prefer?

**Comment by paulvi** [comment]
OK, I see. In some places it is better as it is.

So let it be so. One can select function reference and send  type `(`

**Comment by maxandersen** [comment]
it sounds like sorting is done based on ascii ordering of the string proposal ? That is almost always never the right choice.

If the sorting was done by type of "most likely" + complexity (i.e. sort by function vs method vs matching returntype and then number of parameters) and *then* alphabetically the sort would be what I think most would expect.

But if such sorting is not possible currently then the current mechanism is the best approximation.

**Comment by maxandersen** [comment]
If you look into eclipse JDT or even WTP xml sorting they have extra metadata to provide a relevance score which is then what is sorted by besides just plain asciisort.

Would be a good thing for tern to allow more than just string sorting.

**Comment by angelozerr** [comment]
Yes @maxandersen you are right, but I don't know if tern can support that (in standard).
